docs/design/custom-plugin.md

```diff
@@ -0,0 +1,72 @@
+# Custom Plugin
```

Comment:
Please submit the design doc in a seperated pull request :)

Reply:
The reason is ?

Reply:
> The reason is ?

It's much more friendly for newers to have a better understanding of this feature

Reply:
I mean why need to seperate feature and doc?